docs/root/configuration/listeners/network_filters/generic_proxy_filter.rst

```diff
@@ -0,0 +1,128 @@
+  The property could be any property of the request that indexed by string key.
+* :ref:`request input <envoy_v3_api_msg_extensions.filters.network.generic_proxy.matcher.v3.RequestMatchInput>` and
+  :ref:`request matcher <envoy_v3_api_msg_extensions.filters.network.generic_proxy.matcher.v3.RequestMatcher>`: match the whole request by combining
+  host, path, method and properties in AND semantics. This is used to match multiple fields of the downstream request and avoid complex combinations of
+  host, path, method and properties in the generic matching tree.
+
+.. note::
+   Virtual methods are used to get the request properties in ``Request`` like ``host()``, ``path()``, ``method()``, etc. The developers need to
+   implement these virtual methods and determine what these fields are in their own protocol. It is possible that a protocol does not have some of
+   these fields. For example, the protocol does not have ``host`` field. In this case, the developers could return empty string in the virtual method
+   to indicate the field does not exist in the protocol. The only drawback is that the generic proxy could not match the request by this field.
+
+
+.. literalinclude:: _include/generic_proxy_filter.yaml
+    :language: yaml
+    :linenos:
+    :lineno-start: 21
+    :lines: 21-53
+
+Asynchronous codec API
+----------------------
+
+The generic proxy provides extension point to let the developers to implement specific codec for their own protocol. The codec API is designed to be
+asynchronous to avoid blocking the worker thread. And the asynchronous codec API make it is possible to accelerate the codec by offloading the parsing
+work to specific hardware.
+
+
+Configurable connection
+-----------------------
+
+Different protocols may have different connection lifecycle or connection management. The generic proxy provides additional options to let the developers
+of codec to configure the connection lifecycle and connection management.
+
+
+For example, the developers could configure the upstream connection be bound to the downstream connection or not. If the upstream connection is bound to
+the downstream connection, the upstream connection will have same lifetime with the downstream connection. And the bound upstream connection will be
+used only by the requests come from related downstream connection. This is useful for the protocols that need to keep the connection state.
+
+
+In addition, the developers could operate the downstream connection and upstream connection in the codec directly. This give the developers more control
+of the connection.
+
+
+Example codec implementation
+----------------------------
+
+The community has implemented a :ref:`dubbo codec <envoy_v3_api_msg_extensions.filters.network.generic_proxy.codecs.dubbo.v3.DubboCodecConfig>` based on
+generic proxy. The dubbo codec is a good example to show how to implement a new codec for new protocol because its moderate complexity.
```

Comment:
```suggestion
generic proxy. The dubbo codec is a good example showing how to implement a new codec for new protocol because of its moderate complexity.
```